fdbclient/FileBackupAgent.actor.cpp

```diff
@@ -712,10 +710,15 @@ struct EncryptedRangeFileWriter : public IRangeFileWriter {
 		return std::make_pair(FDB_DEFAULT_ENCRYPT_DOMAIN_ID, FDB_DEFAULT_ENCRYPT_DOMAIN_NAME);
 	}
 
-	static Future<std::pair<int64_t, TenantName>> getEncryptionDomainDetails(
-	    KeyRef key,
-	    Reference<TenantEntryCache<Void>> tenantCache) {
-		return getEncryptionDomainDetailsImpl(key, tenantCache);
+	static Future<std::pair<int64_t, TenantName>> getEncryptionDomainDetails(KeyRef key,
+	                                                                         EncryptedRangeFileWriter* self) {
+		// If tenants are disabled on a cluster then don't use the TenantEntryCache as it will result in alot of
+		// unnecessary cache misses. It is strongly recommended NOT to use snapshot backups with
+		// TenantMode::OPTIONAL_TENANT for now as it can also result in many cache misses
+		bool useTenantCache = self->cx->clientInfo->get().tenantMode != TenantMode::DISABLED &&
+		                      CLIENT_KNOBS->BACKUP_ENCRYPTED_SNAPSHOT_USE_TENANT_CACHE;
```

Comment:
Am still unsure of the intention of using KNOB to control the tenantCache, if testing is the intention then `BUGGIFY` would be good enough. Such a KNOB seems risky especially when cache disable leads to severe performance penalty in all possible scenarios.

Reply:
See response above

Reply:
ditto